### PR TITLE
Code cleanup

### LINKED ISSUE #24: Make HMC and NUTS return extra information
Information such as whether a divergence occurred, the number of integration steps (for NUTS), the value of the energy, etc. can be useful to diagnose sampling issues. Here are some examples:

- momentum
-  acceptance_probability
- is_accepted
-  is_divergent
- energy
- proposal
 - num_integration_steps

### LINKED ISSUE #28: Rename `hmc.kernel` and `nuts.kernel`
The function name `kernel` can be misleading as this function *creates* a new kernel. We should rename it to something along the lines of `build_kernel`.

### LINKED ISSUE #63: Remove reference to JAX in docstring
https://github.com/aesara-devs/aehmc/blob/ede3f376d07c1f1858b8b55fea4a937207913535/aehmc/trajectory.py#L114

### PR DESCRIPTION
**Pull request by rlouf**
In this PR I do some light refactoring: 
1. Updating the docstrings (closes #63)
2.  Return `is_divergent` boolean from HMC transition (closes #24)
3.  Rename s/kernel/new_kernel (closes #28).

### CODE REVIEW

**Comment by rlouf** [comment]
Tests fail for something completely unrelated, same as #61.

**Comment by brandonwillard** [comment]
The bug seen here should be fixed by https://github.com/aesara-devs/aesara/pull/1035.

**Comment by rlouf** [comment]
I ran the tests locally with the branch corresponding to https://github.com/aesara-devs/aesara/pull/1035 and they all pass. We can merge this one if the changes are ok with you.

File: tests/test_step_size.py [hunk 1]
```diff
@@ -32,7 +32,7 @@ def test_dual_averaging_adaptation(init):
     step, logstepsize, logstepsize_avg, gradient_avg, mu = init_fn(init_stepsize)
 
     def one_step(q, logprob, logprob_grad, step, x_t, x_avg, gradient_avg, mu):
-        (*state, p_accept), inner_updates = kernel(
+        (*state, p_accept, _), inner_updates = kernel(
```
Comment (brandonwillard) [inline tests/test_step_size.py:35]: This looks like a potential bug fix that ended up in an unrelated commit.

**Review by brandonwillard — APPROVED** [review]
